include/common.h

```diff
@@ -10,6 +10,10 @@
 typedef unsigned long vm_addr_t;
 typedef unsigned long pm_addr_t;
 
+/* Wrapper for various GCC attributes */
+#define __nonnull(x)    __attribute__((__nonnull__(x)))
+#define __returns_twice __attribute__((__returns_twice__))
```

Comment:
You introduce this macro, but never use it - seems like you forgot to apply it to some function.